kakao/src/main/kotlin/com/agoda/kakao/common/builders/ViewBuilder.kt

```diff
@@ -387,13 +388,15 @@ class ViewBuilder {
     }
 
     /**
-     * Returns view interaction based on all given matchers
+     * Returns view interaction delegate based on all given matchers
      *
-     * @return ViewInteraction
+     * @return ViewInteractionDelegate
      */
-    fun getViewInteraction(): ViewInteraction {
+    fun getViewInteractionDelegate(): ViewInteractionDelegate {
         check(viewMatchers.isNotEmpty()) { "No matchers inside InteractionBuilder" }
-        return Espresso.onView(AllOf.allOf(viewMatchers))
+        return InteractionDelegatesFactory().createViewInteractionDelegate(
```

Comment:
Can you please justify why do we need to use factory pattern here?

Reply:
Factory is the single input point to provide *DelegateInteraction at the whole project. It's so convenient especially when you are refactoring a code.
I can't understand why it's bad?